### ISSUE TITLE
`MockClientState` to `RawMockClientState` conversion ignores `frozen_height`

### ISSUE THREAD

**Issue by rnbguy** [cosmos/ibc-rs#1052]
## Bug Summary

The following implementation ignores [`frozen_height`](https://github.com/cosmos/ibc-rs/blob/main/ibc-testkit/src/testapp/ibc/clients/mock/client_state.rs#L42) of `MockClientState`.

https://github.com/cosmos/ibc-rs/blob/ee92645e050f552bee9acdd17ba6468a624c710c/ibc-testkit/src/testapp/ibc/clients/mock/client_state.rs#L87-L96

So, the serialization-deserialization may not work as expected.

## Details

 For example, the following fails,

```rs
let header = MockHeader::new(Height::new(0, 1).unwrap());
let client_state = MockClientState::new(header).with_frozen_height(Height::min(0));
let raw_client_state = RawMockClientState::from(client_state);
let client_state_back = MockClientState::try_from(raw_client_state).unwrap();
assert_eq!(client_state, client_state_back);
```

We need to update the original [`ibc.mock` definitions on `ibc-proto-rs` repo](https://github.com/cosmos/ibc-proto-rs/blob/da8dbee83ed034fffb028a1966cc4c69ae17f053/definitions/mock/ibc.mock.proto) and fix the `From` implementation in this repo.

## Version

<= v0.49.1

Tracked by https://github.com/cosmos/ibc-rs/tree/feat/refactor-testkit

Will be fixed by #1068

**Comment by rnbguy** [comment]
Note: the same happens with `MockConsensusState` and `RawMockConsensusState` where it ignores the [`commitment root`](https://github.com/cosmos/ibc-rs/blob/main/ibc-testkit/src/testapp/ibc/clients/mock/consensus_state.rs#L18).

**Comment by gr4yha7** [comment]
@rnbguy I can pick this up.

**Comment by rnbguy** [comment]
sorry, I am already working on this on the feature branch https://github.com/cosmos/ibc-rs/tree/feat/refactor-testkit

**Comment by rnbguy** [comment]
closed by #1109
